src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+                case YEARS:
+                    return plusYears(amountToAdd);
+                default:
+                    break;
+            }
+        }
+
+        return (InternationalFixedDate) super.plus(amountToAdd, unit);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    InternationalFixedDate plusYears(final long yearsToAdd) {
+        if (yearsToAdd == 0) {
+            return this;
+        }
+
+        int newYear = (int) Math.addExact(getProlepticYear(), yearsToAdd);
+
+        if (isYearDay()) {
+            return create(newYear, false, true);
+        }
+
+        int day = getDayOfMonth();
+        int newMonth = getCalculatedMonth();
+
+        if (isLeapDay()) {
+            if ((yearsToAdd & 3) == 0) {
+                return create(newYear, true, false);
+            }
+
+            day = 1;
+        }
+
+        return create(newYear, newMonth, day);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate plusMonths(final long months) {
+        if (months == 0) {
+            return this;
+        }
+
+        int day = isYearDay() ? InternationalFixedChronology.DAYS_IN_MONTH : getCalculatedDayOfMonth();
+        int newMonth = (int) Math.addExact(getCalculatedProlepticMonth(), months);
+        int year = newMonth / InternationalFixedChronology.MONTHS_IN_YEAR;
+        newMonth = 1 + (newMonth % InternationalFixedChronology.MONTHS_IN_YEAR);
+
+        return create(year, newMonth, day);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate plusDays(final long daysToAdd) {
+        if (daysToAdd == 0) {
+            return this;
+        }
+
+        return ofEpochDay(toEpochDay() + daysToAdd);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ValueRange range(final TemporalField field) {
+        boolean special = isYearDay() || isLeapDay();
+
+        if (field instanceof ChronoField) {
+            if (isSupported(field)) {
+                ChronoField f = (ChronoField) field;
+
+                switch (f) {
+                    case ALIGNED_DAY_OF_WEEK_IN_MONTH:
+                    case ALIGNED_DAY_OF_WEEK_IN_YEAR:
+                    case DAY_OF_WEEK:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : InternationalFixedChronology.DAY_OF_WEEK_RANGE;
+                    case ALIGNED_WEEK_OF_MONTH:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : InternationalFixedChronology.WEEK_OF_MONTH_RANGE;
+                    case ALIGNED_WEEK_OF_YEAR:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : InternationalFixedChronology.WEEK_OF_YEAR_RANGE;
+                    case DAY_OF_MONTH:
+                        return special ? InternationalFixedChronology.EMPTY_RANGE : InternationalFixedChronology.DAY_OF_MONTH_RANGE;
+                    case DAY_OF_YEAR:
```

Comment:
... the actual range doesn't change for `DAY_OF_YEAR`, when it's leap-day or year-day.